src/ZipkinBundle/Middleware.php

```diff
@@ -122,7 +122,7 @@ public function onKernelTerminate(PostResponseEvent $event)
 
         $statusCode = $event->getResponse()->getStatusCode();
         if ($statusCode > 399) {
-            $span->tag(Tags\ERROR, $statusCode);
+            $span->tag(Tags\ERROR, 'true');
```

Comment:
Related to this change, I guess that there is a conflict between this assignment here, and the one in `ZipkinBundle\Middleware::onKernelException()`, which gets executed before this one...

https://github.com/jcchavezs/zipkin-instrumentation-symfony/blob/989ad9dc858ee02426954b96c651e5912dea81a1/src/ZipkinBundle/Middleware.php#L96-L98

Both assignments use the same key in the span's bagage items.

According to the [docs](https://opentracing.io/docs/overview/tags-logs-baggage/):

> Examples may include tag keys like `db.instance` to identify a database host, `http.status_code` to represent the HTTP response code, **or `error` which can be set to True** if the operation represented by the Span fails.

So maybe we should pick another key for the exception message, assigned to `Tags\Error` in #L96-L98 ?

Reply:
Hi @UrGuardian4ngel I think we are mixing two things here: The docs you are pointing are for `OpenTracing` while this is the zipkin instrumentation for symfony. Either way you are right there is some sort of unconsistency between the exception message and the error tag (status code). @adriancole do you have any input on this, what do other frameworks do?

Reply:
we use the error tag to include a short message, ideally human readable as opposed to an error code, but error code in worst case. For example, "Socket disconnected" "SocketException" "503" are all valid. Try not to embed huge things like stack traces, or things that clarify nothing like "true"